configs/module.go

```diff
@@ -178,16 +179,20 @@ func (m *Module) appendFile(file *File) hcl.Diagnostics {
 	}
 
 	for _, reqd := range file.RequiredProviders {
-		// As an interim *testing* step, we will accept a source argument
-		// but assume that the source is a legacy provider. This allows us to
-		// exercise the provider local names -> fqn logic without changing
-		// terraform's behavior.
+		var fqn addrs.Provider
+		var parseDiags tfdiags.Diagnostics
 		if reqd.Source != "" {
-			// Fixme: once the rest of the provider source logic is implemented,
-			// update this to get the addrs.Provider by using
-			// addrs.ParseProviderSourceString()
+			fqn, parseDiags = addrs.ParseProviderSourceString(reqd.Source)
+			if parseDiags.HasErrors() {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid provider source string",
+					Detail:   parseDiags.Err().Error(),
```

Comment:
Since `ParseProviderSourceString` is already returning a diagnostics here anyway, it's usually best to just pass through the inner diagnostic rather than wrapping like this, or else the result is likely to look pretty weird. (The stringification of a diagnostics as an error is mainly for internal use in our logs, not something intended to be friendly to end-users.)

Since we're already accumulating diagnostics in `diags` here anyway, I guess that would be:

```go
    diags = diags.Append(parseDiags)
```

...which should produce a nice result as long as the diagnostics produced by `ParseProviderSourceString` are of good quality already.